### PR TITLE
Create a branch cache volume for untrusted builds

### PR DESCRIPTION
**Pull request by FintanH**
Let me know if this looks right

### CODE REVIEW

File: linux/etc/buildkite-agent/hooks/environment [hunk 1]
```diff
@@ -92,6 +98,12 @@ then
 else
     # Use kata-containers for isolation
     export DOCKER_RUNTIME=kata-containers
+
+    # Use branch cache
+    declare -r branch_cache_volume="${cache_volume_prefix}_${BUILDKITE_BRANCH}"
```
Comment (geigerzaehler) [inline linux/etc/buildkite-agent/hooks/environment]: For pull requests `BUILDKITE_BRANCH` will be `repo:branch` but `:` is not a valid character for volume names. We need to replace the colon with something.

Comment (FintanH) [reply linux/etc/buildkite-agent/hooks/environment]: Replace all `:` with `-` maybe?

**Comment by FintanH** [comment]
> You can also simplify the DOCKER_CACHE_MOUNT variable: You can remove all volume-opt options and the volume-driver-option. But you need to set the src option to $cache_volume.

Is that in both cases? So they would look like:
```bash
export DOCKER_CACHE_MOUNT="type=volume,src=${cache_volume},dst=/cache,volume-opt=exec=on"
```

**Comment by geigerzaehler** [comment]
> Is that in both cases? So they would look like:
> 
> ```shell
> export DOCKER_CACHE_MOUNT="type=volume,src=${cache_volume},dst=/cache,volume-opt=exec=on"
> ```

You can even drop the `volume-opt=exec=on`. But we should retain `volume-nocopy=true` for the trusted build and also add that option to the untrusted builds.

**Comment by FintanH** [comment]
It is weird that the refquota is not the usual 8 gigs though right? Could it be that the branch name is messing with things?

**Comment by kim** [comment]
Ok, so, you're looking at the wrong volume: try passing `refquota` explicitly [here](https://github.com/oscoin/ci/blob/master/linux/etc/buildkite-agent/hooks/command#L127)

**Comment by FintanH** [comment]
> Ok, so, you're looking at the wrong volume: try passing refquota explicitly here

Ah because it's running under the `command` hook :man_facepalming:

**Comment by geigerzaehler** [comment]
> I don't think this is the right fix -- the volume is explicitly anonymous, and this should stay this way if possible. The mentioned build errors on the `img` step, so is also unrelated.

Of course! You’re right.

**Comment by FintanH** [comment]
> Ok, so, you're looking at the wrong volume: try passing refquota explicitly here

Thanks, I made that change. Is that what you meant? :crossed_fingers:

**Comment by FintanH** [comment]
Is there some way to test these kind of changes? :thinking:

**Comment by kim** [comment]
@FintanH One way is to spin up a clown instance using https://github.com/oscoin/ci/blob/master/linux/gce/provision.sh and target it with a build. Another way is to get #54 out, which will eventually allow to spin up the stack in a local VM.

**Comment by FintanH** [comment]
@kim it seems that I'd need gce credentials for auth. I'll merge this for now and look at your Rusty code :grin:

**Pull request merged**